### PR TITLE
worker: support shutdown

### PR DESCRIPTION
**Pull request by BusyJay**
Sometimes we need to do some cleanup when worker is shutdown.

@siddontang @disksing @zhangjinpeng1987 PTAL

### CODE REVIEW

**Comment by siddontang** [comment]
LGTM

**Comment by ngaut** [comment]
LGTM

File: src/util/worker/mod.rs [hunk 1]
```diff
@@ -144,6 +150,7 @@ fn poll<R, T>(mut runner: R, rx: Receiver<Option<T>>, counter: Arc<AtomicUsize>,
         let t = rx.recv();
         match t {
             Ok(Some(t)) => buffer.push(t),
+            Ok(None) => break,
```
Comment (disksing) [inline src/util/worker/mod.rs]: No need shutdown for error?

**Comment by disksing** [comment]
LGTM

**Pull request merged**